VALIDATION-SCHEMA.md

```diff
@@ -0,0 +1,116 @@
+input is sparse this model will produce a sparse matrix, and if the input is dense it will output dense.  
+
+### number_of_columns
+The constraint for number of columns allows specifying if a specific number of columns are required or if the model has
+some maximum or minimum number of columns.  For time consuming models specifying a maximum number of columns can help
+keep performance reasonable.  Number of columns allows multiple entries to create ranges of values that are allwoed.  
+Note that some conditions only allow a single entry.
+
+- EQUALS: The number of columns in the dataframe will exactly match the value.  No additional conditions allowed.
+- IN:  Multiple possible acceptable values are possible.  No additional conditions allowed.
+- NOT_EQUALS: The number of columns must not be the specified value.
+- GREATER_THAN: The number of columns in the dataframe must be greater than the value provided.
+- LESS_THAN: The number of columns in the dataframe must be less than the value provided.
+- NOT_GREATER_THAN: The number of columns in the dataframe must be less than or equal to the value provided.
+- NOT_LESS_THAN: The number of columns in the dataframe must be greater than or equal to the value provided.
+
+### contains_missing
+The `contains_missing` field specifies if a model can accept missing data, or if missing data can be passed on from the
+model in the output.  A condition of EQUALS will always be used. 
+For the input requirements the following values are allowed:
+- FORBIDDEN: The model cannot accept missing values/NA in the input dataframe. 
+- SUPPORTED: The model is capable of dealing with missing values.
+
+For output requirements the following values are allowed:
+- NEVER: The model will never output missing values.
+- DYNAMIC: The model might output missing values.
+
+## Running checks locally
+
+When running `drum fit` or `drum push` the full set of validation will automatically be run.  The first step of verification
+checks the validity of the file format.  Any format issues must be addressed before the model can be trained locally or 
+on DataRobot.  After format validation the input dataset used for fit is compared against the supplied input_requirements
+specifications.  Following model training the output of the model will be compared to the output_requirements.  
+
+### ignoring validation
+During development of a model it might be useful to disable the validation.  In that case the `--disable-strict-validation`
+may be used to ignore any errors.  
+
+## Example typeSchema
+This example would be in addition to other required fields in model_metadata.yaml.  
+```yaml
+-typeSchema:
+  - input_requirements:
+      - field:  data_types
+        condition: NOT_IN
+        value: 
+          - TXT
+          - IMG
+          - DATE
+      - field: sparse
+        condition: EQUALS
+        value:  SUPPORTED
```

Comment:
this is pretty strange. Why not just sparse SUPPORTED? why need the EQUALS?

Reply:
i'm in favor of keeping the `EQUALS` because it means that all of these type schemas keep the same format.  no one has to guess which type schemas need a condition, you just _always have a condition_

Reply:
I also prefer a consistent format instead of having special cases.